### ISSUE TITLE
[feature] Unique comment identifier

### ISSUE THREAD

**Issue by michelcaradec** [d-koppenhagen/vscode-code-review#75]
# 🧩 Feature

## Description

As of today, a comment is identified by the **filename** and the **selection**.
I suggest to work on it to provide a unique identifier (i.e. a GUID).
This will allow:
- Multiple comments on the same selection.
- To assign a comment to a file (vs a selection).
- More reliable comment lookup.

What do you think?

**Comment by d-koppenhagen** [comment]
I think it's a Good idea but it would cause a breaking change won't it? 

So how should existing review files be handled? Probably the information would just be another CSV column containing a uuid?
If the format is change there should be an automatic migration for existing files. Also all export function should be checked carefully (Gitlab , github, JIRA, etc.)

**Comment by michelcaradec** [comment]
The idea is to add an extra field (the Id) at the end of each row (as an extra column). 
At read time, the Id will be read if it exists (for new files). 
At search time, the Id will be used if present. If not, a fallback will use the existing method based on file name and selection. 

Regarding migration, the best place to do it is probably when the csv file is read during extension initialization (in this case, the fallback method mentioned above would become useless). 

The export shouldn't be a problem, as no field will be removed. But the parsing must be checked.

To summarize, I don't see any difficulty. But all edge cases will have to be tested.

**Comment by d-koppenhagen** [comment]
I think in terms of consistency a migration would be the better approach what do you think?
each unfilled id column for each row will be filled with a uuid, that's generated (the `uuid` NPM package can probably be used here).
As I am quite busy this days, I would be very happy to receive a PR.

**Comment by michelcaradec** [comment]
I started working on the implementation this weekend, with migration oriented design (by far the simplest and safest way of doing).
It's working as expected by now. 
Note this required some refactoring to centralize storage logic (I will describe it in the PR). 
I want to check everything again before submitting the PR.

**Comment by d-koppenhagen** [comment]
Awesome 😎 that sounds great! Thanks for your great ideas and contributions!

**Comment by michelcaradec** [comment]
PR submitted!
